### PR TITLE
update yubikey documentation

### LINKED ISSUE #1506: Yubikey enrollment with yubikey personalization tool
It is possible to enroll yubikeys with the yubikey personalization tool.
The tool can initialize multiple yubikeys and write the seeds to a file.

The documentation should describe.
1. which settings you need to do in the tool (preferably HOTP, no identifier) 
2. configure the columns of the csv file
3. save the file
4. import the file to privacyIDEA.

Including screenshots.

We already have some links to the peronalization tool. references should be added in the documentation. https://privacyidea.readthedocs.io/en/latest/configuration/tokens/yubikey.html?highlight=yubikey#using-the-yubikey-personalization-gui

# Why would we do this?

We prefer the initialization with the privacyidea admin client. It is easier in mass production but a bit difficult to set up. Using the personalization tool gets you a quicker start, especially on windows. (It will be more complicated later, but it lowers the bar for newbies.)

### LINKED ISSUE #2318: Update the docs about yubikey enrollment
We should update the docs about yubikey enrollment.
https://privacyidea.readthedocs.io/en/latest/tokens/tokentypes/yubikey.html?highlight=yubikey#using-the-yubikey-personalization-gui

Especially on windows is an easy way to also mass enroll yubikeys with the personalization tool and then import the seed file to privacyIDEA.

We can add this as two alternatives under 6.3.1.25.1.

1. Mass enroll
2. AES single token enroll

To do so, we can use these two screenshots.
yubikey-personalization-settings
yubikey-personalization-windows

### PR DESCRIPTION
**Pull request by laclaro**
This updates the yubikey documentation as requested in #2318.

closes #1506
closes #2318

### CODE REVIEW

File: doc/tokens/tokentypes/yubikey.rst [hunk 1]
```diff
@@ -3,22 +3,33 @@
 Yubikey
 -------
 
-.. index:: Yubikey, Yubico AES mode
+.. index:: Yubikey, Yubico AES mode, Yubikey OATH-HOTP mode
 
-The Yubikey is initialized with privacyIDEA and works in Yubicos own AES mode.
-It outputs a 44 character OTP value, consisting of a 12 character prefix and
+The Yubikey can be used with privacyIDEA in Yubico's own AES mode (*Yubico OTP*)
+or in the OATH-HOTP mode which is for privacyIDEA a HOTP Token.
```
Comment (cornelinux) [inline doc/tokens/tokentypes/yubikey.rst:9]: I would rephrase this and talk of a token type rather then a token:

    or in the OATH-HOTP mode which is an HOTP token type for privacyIDEA.

File: doc/tokens/tokentypes/yubikey.rst [hunk 2]
```diff
@@ -3,22 +3,33 @@
...
+
+In AES mode it outputs a 44 character OTP value, consisting of a 12 character prefix and
 a 32 character OTP. But in contrast to the :ref:`yubico_token` Cloud
 mode, in this mode the secret key is contained within the token and your own
-privacyIDEA installation.
+privacyIDEA installation. In OATH-HOTP mode, it just prints a series of 6 or 8 digits.
```
Comment (cornelinux) [inline doc/tokens/tokentypes/yubikey.rst:14]: The problem is, that the OATH-HOTP mode does not belong here.
Since this section deals about the tokentype "yubikey". Which **is** the the AES-mode.

Note: This section here does not deal with *all* possible yubikey modes, so it should probably refer for this to other sections.

We probably should restructure the lines starting with line 22.

**Review by cornelinux — CHANGES_REQUESTED** [review]
We need to restructre this.

There are tokentypes, that are described in tokentypes/... and means for enrollment, which are specific to the Yubikey hardware.

``tokentypes/yubikey.rst`` should only deal with the AES mode, since this **is** the AES mode, which in privacyIDEA is the *yubikey* tokentype. 

``tokentypes/hotp.rst`` can also be a Yubikey hardware. This sesction already mentions the Yubikey hardware. 

I think it is best, if both tokentypes have refs to an additional section, that deals with the means to enroll a Yubikey hardware - either in HOTP mode (hotp tokentyp) or in AES mode (yubikey tokentype).

**Comment by laclaro** [comment]
The recent commit https://github.com/privacyidea/privacyidea/pull/2595/commits/ca922338784afd8a7846b457cadc8ec3f420db87 moves the documentation of enrollment of Yubikey devices to workflows_and_tools/enrollment_tools and links the section wherever needed.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+The Yubikey can be used with privacyIDEA in Yubico's own AES mode (*Yubico OTP*),
+in the HOTP mode (*OATH-HOTP*) or the seldomly used static password mode.
+
+This section describes several tools which can be used to initialize and enroll a
+Yubikey with privacyIDEA.
+
+If not using the :ref:`yubico_token` mode, the Yubikey has to be initialized with privacyIDEA
+to bring the token secrets to the system.
+
+privacyIDEA ships several tools to (mass-)enroll Yubikeys in AES mode (Yubikey Token) or HOTP mode (HOTP Token).
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:17]: It think there are not that many. I would probably simply drop "several".

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+client [#privacyideaadm]_, to initialize the Yubikeys. You can use the mass enrollment, which
+eases the process of initializing a whole bunch of tokens.
+
+Run the command like this::
+
+   privacyidea -U https://your.privacyidea.server -a admin token \
+   yubikey_mass_enroll --yubimode YUBICO
+
+This command initializes the token and stores the AES secret and prefix
+in privacyIDEA, so the token is immediately useful. You can enroll Yubikeys
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:36]: "immediately useful" sounds awkward.

The token now exits in privacyIDEA. The admin still has to assign the token to a user.
So I would do s.th. like.

"This command initializes the token and stores the AES secret and prefix
in privacyIDEA. The administrator can now assign the token to a user."

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+in HOTP mode by removing the option ``--yubimode``.
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:37]: Actually it is ``--yubimode HOTP``, which is default.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+You can also initialize the Yubikey with the official Yubico personalization GUI
+[#ykpers]_ and use the obtained secret to enroll the Yubikey with privacyIDEA.
+For both AES (Yubico OTP) and OATH-HOTP mode, there are two possibilities to initialize
+the Yubikey with privacyIDEA.
+
+Single token enrollment
+=======================
+
+To initialize a single Yubikey in AES mode (Yubico OTP) use the *Quick* button and
+copy the shown secret labeled with "Secret Key (16 bytes Hex)" to the field *OTP Key*
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:63]: I think it is rather "displayed secret" than "shown secret".

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+   :width: 500
+
+   *Initialize a Yubikey in AES mode (Yubikey OTP)*
+
+.. figure:: images/enroll_yubikey.png
+   :width: 500
+
+   *Enroll a Yubikey AES mode token in privacyIDEA*
+
+In the field "Test Yubikey" push the Yubikey button. This will determine the
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:76]: The understanding of the Yubikey button is difficult for some users.
I think the best word to be used (I think Yubico uses it themselves) is "touch the Yubikey button".

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+The process is similar for the HOTP mode. You have to deselect *OATH Token Identifier*.
+Copy the shown secret to the HOTP :ref:`hotp_token_enrollment` form in privacyIDEA.
+
+.. figure:: images/ykpers-quick-initialize-oath-hotp.png
+   :width: 500
+
+   *To initialize a single Yubikey in HOTP mode, deselect OATH Token Identifier.*
+
+.. note::
+    In the case of HOTP mode privacyIDEA does not know that the token is a Yubikey.
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:97]: "does not know" sounds a bit strange. I would rather call it like

"In the case of HOTP mode privacyIDEA can not necessarily distinguish a Yubikey in HOTP mode from a smartphone App in HOTP mode. Using the above mentioned mass-enrollment, the token serial number is used to distinguish these tokens."

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+Yubikey Personalization GUI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also initialize the Yubikey with the official Yubico personalization GUI
+[#ykpers]_ and use the obtained secret to enroll the Yubikey with privacyIDEA.
+For both AES (Yubico OTP) and OATH-HOTP mode, there are two possibilities to initialize
+the Yubikey with privacyIDEA.
+
+Single token enrollment
+=======================
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:60]: I think this is the wrong header identifier. It should be a line of dots.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+   *To initialize a single Yubikey in HOTP mode, deselect OATH Token Identifier.*
+
+.. note::
+    In the case of HOTP mode privacyIDEA does not know that the token is a Yubikey.
+    To set an identifying token serial, consider to use the advanced mode with an
+    appropriate file output (see below).
+
+Mass enrollment
+===============
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:102]: This should be a line of dots, since "====" usually is a headline above "------", which is a headline above "~~~~~"....

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+To initialize one or more Yubikeys and to write the configuration to an output file use the
+*Advanced* button. The configured secrets are imported afterwards in the WebUI. After plugging
+the Yubikey, it will be detected and the selected configuration slot is initialized upon
+*Write Configuration* as before. You will be prompted to select an output file name.
+The secrets are written to this file in the format set in *Settings* -> *Log configuration output*.
+We recommend to select *Yubico format* since here privacyIDEA is able to detect the Yubikey mode and
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:109]: We do not read "Yubico format"!
It mus be "Flexible format".

For HOTP it should be:

    UBOM{serial}_{configSlot},{secretKeyTxt},HOTP,{hotpDigits}  (tested)

For AES mode ist should:

    UBAM{serial}_{configSlot},{secretKeyTxt},yubikey  (Not tested!)

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+sets the serial accordingly. PSKC format is also supported upon import. You may also use a custom
+output configuration to set custom token serials upon import using :ref:`import_oath_csv`.
```
Comment (cornelinux) [inline doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:111]: Please check, if the rest of the seed-file description fits my above comment.

**Review by cornelinux — CHANGES_REQUESTED** [review]
Please note my wording comments.

There is an error in the seed-file formats.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst (hunk 5 above)
Comment (laclaro) [reply doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:37]: true, I'll put that explicitly.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst (hunk 11 above)
Comment (laclaro) [reply doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst:106]: We do read yubico format, for hotp and AES mode. But we can use flexible format to prepend a different string for the serial. I reworked this section.

File: doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst (hunk 12 above)
Comment (laclaro) [reply doc/workflows_and_tools/enrollment_tools/yubikey_enrollment_tools.rst]: done

**Review by laclaro — COMMENTED** [review]
Please check again.

**Review by cornelinux — APPROVED** [review]
